### PR TITLE
Add Windows build instructions to how-to-build.md

### PR DESCRIPTION
**Pull request by LanceHsun**
### What changes were proposed in this pull request?

(Please outline the changes and how this PR fixes the issue.)
This pull request adds instructions for building Gravitino on Windows using the Windows Subsystem for Linux (WSL). The new section provides a detailed step-by-step guide to setting up the necessary environment and dependencies on Windows to successfully build and run Gravitino. Additionally, it includes specific instructions provided by minghuang and docs written by Junshen Tao, which have been tested and proven effective for building the project.
### Why are the changes needed?
These changes are needed to provide Windows users with clear and concise instructions on how to build Gravitino using WSL. This expands the accessibility of the project to a wider audience who may be using Windows as their primary development environment. It addresses the need for comprehensive documentation that covers all major operating systems, including Windows.

Fixes: # (No specific issue)

### Does this PR introduce _any_ user-facing change?

No user-facing APIs or properties were changed. The addition is purely documentation, aimed at helping developers set up their environment on Windows using WSL.

### How was this patch tested?
The documentation changes were reviewed for accuracy and clarity. The steps outlined were followed to ensure they work as described, using an environment set up with Windows 11 and Ubuntu 22.04 on WSL. The specific instructions provided by minghuang were used to successfully build the project, confirming their effectiveness.

### CODE REVIEW

**Comment by yuqi1129** [comment]
Can you create a fork of Gravitino and use your own repo to create the PR?

File: docs/how-to-build.md [hunk 1]
```diff
@@ -151,3 +155,197 @@ This software is licensed under the Apache License version 2."
...
+
+2. Add the following lines at the end of the file. Replace `/usr/lib/jvm/java-11-openjdk-amd64` with your actual Java installation path:
+
+    ```sh
+    export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
+    export PATH=$PATH:$JAVA_HOME/bin
+    ```
+
+3. Save and quit in vim using `:wq`.
+
```
Comment (jtao1) [inline docs/how-to-build.md:200]: We should add 

4. Run `source ~/.bashrc` to which updates the .bashrc file in your current shell session.

File: docs/how-to-build.md [hunk 2]
```diff
@@ -151,3 +155,197 @@ This software is licensed under the Apache License version 2."
...
+Editing the `~/.bashrc` file allows you to set environment variables that will be available in every terminal session. Setting `JAVA_HOME` and updating `PATH` ensures that your system uses the correct Java version for development.
+
+### Install Docker
+
+**On Ubuntu (WSL):**
+
+Refer to chatGPT for specific instructions. Here are the steps:
```
Comment (mchades) [inline docs/how-to-build.md:207]: Have these steps been validated locally?

Comment (LanceHsun) [reply docs/how-to-build.md:207]: yes, I made it to install docker on WSL.

File: docs/how-to-build.md (hunk 1 above)
Comment (LanceHsun) [reply docs/how-to-build.md:200]: ok, thanks Tao!

**Comment by LanceHsun** [comment]
> Can you create a fork of Gravitino and use your own repo to create the PR?

ok, I will close current pr and create another one. Thanks Qian!

**Pull request closed without merging**